delivery_total_weight_from_packaging/models/stock_quant_package.py

```diff
@@ -1,32 +1,43 @@
 # Copyright 2021 Camptocamp SA
 # License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl)
 
-from odoo import api, models
+from odoo import api, fields, models
 
 
 class StockQuantPackage(models.Model):
     _inherit = "stock.quant.package"
 
+    shipping_weight = fields.Float(compute="_compute_shipping_weight", readonly=False)
+
     @api.depends("quant_ids")
     @api.depends_context("picking_id")
     def _compute_weight(self):
         # Override method from `delivery` module to compute a more accurate
         # weight by including the weight of the packaging
         # NOTE: code copied/pasted and adapter from `delivery`
         for package in self:
-            weight = 0.0
-            if self.env.context.get("picking_id"):
-                current_picking_move_line_ids = self.env["stock.move.line"].search(
-                    [
-                        ("result_package_id", "=", package.id),
-                        ("picking_id", "=", self.env.context["picking_id"]),
-                    ]
+            package.weight = package._get_weight_from_packaging()
+
+    @api.depends("quant_ids")
+    @api.depends_context("picking_id")
+    def _compute_shipping_weight(self):
+        for package in self:
+            package.shipping_weight = package._get_weight_from_packaging()
```

Comment:
Should we not compute the shipping weight based on the delivery packaging (`package.packaging_id`)? Here I suppose that `_get_weight_from_packaging()` and `product.get_total_weight_from_packaging()` compute the weight from storage packaging, not delivery ones.

Reply:
hmm good point! I'll check!

Reply:
@sebalix answer: no. The weight on the delivery package is just the max weight allowed. Also, in theory, the weight of the pallet is already considered in the product packaged on its pallet packaging type. We might have issues w/ parcels but we'll tackle it in another round.

Reply:
There is no weight on a delivery package. And the weight from packaging can already include the weight of the packaging. It all depend if you repack in a bigger packaging. That is a very complex other problem that should be addressed here.